google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -35,61 +33,85 @@ char const* ConsumeArg(int& argc, char* argv[]) {
   return result;
 }
 
+std::string command_usage;
+
 void PrintUsage(int argc, char* argv[], std::string const& msg) {
   std::string const cmd = argv[0];
   auto last_slash = std::string(cmd).find_last_of('/');
   auto program = cmd.substr(last_slash + 1);
   std::cerr << msg << "\nUsage: " << program << " <command> [arguments]\n\n"
-            << "Examples:\n";
-  for (auto const& example :
-       {"read-object <bucket-name> <object-name>",
-        "insert-object <object name> <object contents>"}) {
-    std::cerr << "  " << program << " " << example << "\n";
+            << "Commands:\n"
+            << command_usage << std::endl;
+}
+
+void ListObjects(google::cloud::storage::Client client, int& argc,
+                 char* argv[]) {
+  if (argc < 2) {
+    throw Usage{"list-objects <bucket-name>"};
   }
-  std::cerr << std::flush;
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [list objects] [START storage_list_files]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name) {
+    for (gcs::ObjectMetadata const& meta : client.ListObjects(bucket_name)) {
+      std::cout << "bucket_name=" << meta.bucket()
+                << ", object_name=" << meta.name() << std::endl;
+    }
+  }
+  //! [list objects] [END storage_list_files]
+  (std::move(client), bucket_name);
 }
 
-//! [insert object]
-void InsertObject(gcs::Client client, int& argc, char* argv[]) {
+void InsertObject(google::cloud::storage::Client client, int& argc,
+                  char* argv[]) {
   if (argc < 3) {
     throw Usage{
         "insert-object <bucket-name> <object-name> <object-contents (string)>"};
   }
   auto bucket_name = ConsumeArg(argc, argv);
   auto object_name = ConsumeArg(argc, argv);
   auto contents = ConsumeArg(argc, argv);
-  auto meta =
-      client.InsertObject(bucket_name, object_name, std::move(contents));
-  std::cout << "The new object metadata is " << meta << std::endl;
+  //! [insert object] [START storage_upload_file]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name, std::string object_name,
+     std::string contents) {
+    gcs::ObjectMetadata meta =
+        client.InsertObject(bucket_name, object_name, std::move(contents));
+    std::cout << "The new object metadata is " << meta << std::endl;
```

Comment:
Change "The new object metadata is":
to: "The file was uploaded. The new object metadata is: "

Request is to acknowledge the upload.

Reply:
Fixed.